README.Docker.md

```diff
@@ -0,0 +1,246 @@
+servers for example, where only `clamscan` is installed on the host, and
+`freshlamd` is managed in a docker container.
+
+Note, that running `freshclamd` separated from `clamd` is less recommended,
+unless the `clamd` socket is shared with `freshclamd` as `freshclamd` would not
+be able to inform `clamd` of database updates.
+
+
+### Dockerized clamscan
+To run `clamscan` in a docker container, the docker container can be invoked
+as:
+```console
+docker run \
+       --interactive \
+       --rm \
+       --tty \
+       --volume '/path/to/scan:/scandir' \
+       "myclamav" clamscan /scandir
+```
+This would pull the virus database using `freshclam`, run the scan and cleanup.
+
+Note, that this is very inefficient when done often, as the complete database is
+downloaded on each invocation.
+
+
+### clamdscan
+As with `clamscan` `clamdscan` can also be run when installed on the host, by
+connecting to the dockerized `clamd`. This can be done by either pointing
+`clamdscan` to the exposed TCP/UDP port or unix socket.
+
+
+### Dockerized clamdscan
+Running both `clamd` and `clamdscan` is also easily possible, as all that is
+needed is the shared socket between the two containers. The only cavaet here
+is to mount the files to be scaned in the container that is expected to run
+`clamdscan`.
+
+For example:
+```console
+docker run \
+       --interactive \
+       --rm \
+       --tty \
+       --volume '/path/to/scan:/scandir' \
+       --volume '/var/lib/docker/data/clamav/sockets/:/run/clamav/'
+       "myclamav" clamdscan /scandir
+```
+
+
+## Controlling the container
+The ClamAV container actually runs both `freshclam` and `clamd` by default.
+Optionally available to the container is also ClamAV's milterd. To control the
+behavior of the services started within the container, the following flags can
+be passed to the `docker run` command with the `--environment` parameter.
+
+* CLAMAV_NO_CLAMD [true|**false**] Do not start clamd (default: start clamd)
+* CLAMAV_NO_FRESHCLAMD [true|**false**] Do not start freshclamd (default: start freshclamd)
+* CLAMAV_NO_MILTERD [**true**|false] Do not start milterd (default: no milterd)
+* CLAMD_STARTUP_TIMEOUT [integer] Seconds to wait for clamd to start (default: 1800)
+* FRESHCLAM_CHECKS [integer] Freshclam daily update frequency (default: once per day)
+
+So to additionally also enable milterd, the following flag can be added:
+```console
+       --environment 'CLAMAV_NO_MILTERED=false'
+```
+
+Further more, all of the configuration files that live in `/etc/clamav/` can be
+overridden by doing a volume-mount to the file or entire directory. The
+following argument can be added for this purpose. The example uses the entire
+configuration directory, but this can be supplied multiple times if individual
+files deem to be replaced.
+```console
+       --volume '/full/path/to/clamav/:/etc/clamav'
+```
+
+Note, that even when disabling `flreshclamd`, freshclam will always run at least
```

Comment:
Notes, disclaimers, and tips look really good if you do them like this:

```markdown
> _Note_: Even when disablling ...
```

Eg:
> _Note_: Even when disabling ...

Reply:
I like that, never thought of it as such!